### PR TITLE
 Laboratory work #2, Maxim Glukhankov - 22FPL1

### PR DESCRIPTION
**Pull request by glukhankov**

### CODE REVIEW

File: lab_2_tokenize_by_bpe/main.py [hunk 1]
```diff
@@ -2,12 +2,10 @@
 Lab 2
 BPE and machine translation evaluation
 """
-import json
-import math
 
 
 def prepare_word(
-    raw_word: str, start_of_word: str | None, end_of_word: str | None
+        raw_word: str, start_of_word: str | None, end_of_word: str | None
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:8]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 2]
```diff
@@ -29,7 +27,7 @@ def prepare_word(
 
 
 def collect_frequencies(
-    text: str, start_of_word: str | None, end_of_word: str
+        text: str, start_of_word: str | None, end_of_word: str
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:30]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 3]
```diff
@@ -55,7 +53,7 @@ def collect_frequencies(
 
 
 def count_tokens_pairs(
-    word_frequencies: dict[tuple[str, ...], int]
+        word_frequencies: dict[tuple[str, ...], int]
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:56]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 4]
```diff
@@ -78,7 +76,7 @@ def count_tokens_pairs(
 
 
 def merge_tokens(
-    word_frequencies: dict[tuple[str, ...], int], pair: tuple[str, str]
+        word_frequencies: dict[tuple[str, ...], int], pair: tuple[str, str]
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:79]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 5]
```diff
@@ -107,7 +105,7 @@ def merge_tokens(
 
 
 def train(
-    word_frequencies: dict[tuple[str, ...], int] | None, num_merges: int
+        word_frequencies: dict[tuple[str, ...], int] | None, num_merges: int
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:108]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 6]
```diff
@@ -146,7 +144,7 @@ def train(
 
 
 def get_vocabulary(
-    word_frequencies: dict[tuple[str, ...], int], unknown_token: str
+        word_frequencies: dict[tuple[str, ...], int], unknown_token: str
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:147]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 7]
```diff
@@ -242,24 +219,14 @@ def load_vocabulary(vocab_path: str) -> dict[str, int] | None:
...
-    return vocab
 
 
 def encode(
-    original_text: str,
-    vocabulary: dict[str, int] | None,
-    start_of_word_token: str | None,
-    end_of_word_token: str | None,
-    unknown_token: str,
+        original_text: str,
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:225]: revert

File: lab_2_tokenize_by_bpe/main.py [hunk 8]
```diff
@@ -299,36 +246,17 @@ def collect_ngrams(text: str, order: int) -> list[tuple[str, ...]] | None:
...
-    n_grams = []
-    for index in range(len(text) + 1 - order):
-        n_grams.append(tuple(text[index: index + order]))
-
-    return n_grams
 
 
 def calculate_precision(
-    actual: list[tuple[str, ...]], reference: list[tuple[str, ...]]
+        actual: list[tuple[str, ...]], reference: list[tuple[str, ...]]
```
Comment (artyomtugaryov) [inline lab_2_tokenize_by_bpe/main.py:252]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
You implementation is absolutelly the same as in main branch. It is very strange.

File: lab_2_tokenize_by_bpe/main.py (hunk 1 above)
Comment (glukhankov) [reply lab_2_tokenize_by_bpe/main.py]: done

File: lab_2_tokenize_by_bpe/main.py (hunk 2 above)
Comment (glukhankov) [reply lab_2_tokenize_by_bpe/main.py]: done

File: lab_2_tokenize_by_bpe/main.py (hunk 3 above)
Comment (glukhankov) [reply lab_2_tokenize_by_bpe/main.py]: done

File: lab_2_tokenize_by_bpe/main.py (hunk 4 above)
Comment (glukhankov) [reply lab_2_tokenize_by_bpe/main.py]: done

File: lab_2_tokenize_by_bpe/main.py (hunk 5 above)
Comment (glukhankov) [reply lab_2_tokenize_by_bpe/main.py]: done

File: lab_2_tokenize_by_bpe/main.py (hunk 8 above)
Comment (glukhankov) [reply lab_2_tokenize_by_bpe/main.py]: all done

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**